cmd/limactl/shell.go

```diff
@@ -49,6 +50,7 @@ func newShellCommand() *cobra.Command {
 
 	shellCmd.Flags().String("shell", "", "shell interpreter, e.g. /bin/bash")
 	shellCmd.Flags().String("workdir", "", "working directory")
+	shellCmd.Flags().Bool("restart-ssh", false, "restart the SSH connection")
```

Comment:
This is definitely bike-shedding now, but the `--restart-ssh` name somehow doesn't vibe with me. It feels more like an implementation detail than expressing what the user wants.

I think both `--restart-session` or even `--reconnect` would work better.

Another reason is that I think eventually we will use a different transport depending on the driver. E.g. with WSL2 it makes sense to execute commands via `wsl -d $DISTRO` instead of `ssh`. So using `ssh` in the option name would feel wrong in that situation.